### ISSUE TITLE
Weird result when using query instead of prep_exec

### ISSUE THREAD

**Issue by wqfish** [blackbeam/mysql_async#107]
When debugging an issue in my application that uses `mysql_async`, I ran into a problem where I get different result by calling `query` and `prep_exec`.

I have reduced the program to the following to reproduce the issue:

Cargo.toml: https://gist.github.com/wqfish/7dddb18da14682556e5c33639eeb7b20
main.rs: https://gist.github.com/wqfish/c4a69477b90880c2c5731ce8f90d4747

The content of the db is pretty simple. The db `mydb` has a single table `my_table` that has two rows in it: https://gist.github.com/wqfish/21125e61fc305ba2958083fc795dbef7

### The problem

With this program I can successfully load the two rows from mysql. However, if I comment out line 58 in `main.rs` and use line 57 instead, I only get one of the two rows. So changing `prep_exec` to `query` caused the result to be different.

Does this feel like a bug or I'm missing something?

**Comment by wqfish** [comment]
Another way to demonstrate the problem is the following:

Let's just use `query` instead of `prep_exec`, so I can not get both rows with the above program. However if I make the following change to the code

```
diff --git a/main.rs b/main.rs
index 5c86d32..8c3ee75 100644
--- a/main.rs
+++ b/main.rs
@@ -53,7 +53,7 @@ async fn main() -> Result<(), mysql_async::error::Error> {
 
     let conn = conn.drop_query(r"USE mydb").await.unwrap();
 
-    let q = "SELECT b, c, d, e FROM my_table";
+    let q = "SELECT a, b, c, d, e FROM my_table";
     let result = conn.query(q).await.unwrap();
     // let result = conn.prep_exec(q, ()).await.unwrap();
 
@@ -70,7 +70,7 @@ async fn main() -> Result<(), mysql_async::error::Error> {
 
     let (_ /* conn */, loaded_structs) = result
         .map_and_drop(|row| {
-            let (b, c, d, e): (_, _, _, _) = mysql_async::from_row(row);
+            let (a, b, c, d, e): (u64, _, _, _, _) = mysql_async::from_row(row);
             MyStruct { b, c, d, e }
         })
         .await
```

(Basically I select one more column from the DB.) Then I get both rows.

**Comment by wqfish** [comment]
I spent some time debugging and tracked down to [this code](https://github.com/blackbeam/mysql_async/blob/2e19b2ffff4b7262b30e4c2da411d93f07594f8b/src/queryable/query_result/mod.rs#L116) where we parse the packet to determine if there are more results. It seems that the code returned `None` when it's supposed to return `Some`. I don't really know the format of the packets to debug further.

**Comment by blackbeam** [comment]
Hi. Thanks for report!

Could you please test this using the latest commit on master?

**Comment by wqfish** [comment]
@blackbeam Yeah it seems to be working now on latest master. Thanks for the quick fix!

**Comment by wqfish** [comment]
Before I close this issue, do you mind explaining why https://github.com/blackbeam/rust_mysql_common/pull/17 fixes this? I don't know much about mysql and all I can tell is that I'm not using floats in my code here.

**Comment by blackbeam** [comment]
> Before I close this issue, do you mind explaining why blackbeam/rust_mysql_common#17 fixes this?

Oh. No, your problem is fixed by [8dd31c1][1].

The blackbeam/rust_mysql_common#17 is just another breaking change that I wanted to include into mysql_common v0.21.0 release.

[1]: https://github.com/blackbeam/rust_mysql_common/commit/8dd31c1246d6ed30702c1ca0272f19610b546ed7

**Comment by wqfish** [comment]
Oh that makes sense. Thanks!
